api/lib/infrastructure/repositories/assessment-repository.js

```diff
@@ -109,14 +109,14 @@ module.exports = {
       .then((assessment) => bookshelfToDomainConverter.buildDomainObject(BookshelfAssessment, assessment));
   },
 
-  completeByAssessmentId(assessmentId) {
-    return this.updateStateById({ id: assessmentId, state: Assessment.states.COMPLETED });
+  completeByAssessmentId(domainTransaction, assessmentId) {
+    return this.updateStateById(domainTransaction.knexTransaction, { id: assessmentId, state: Assessment.states.COMPLETED });
```

Comment:
As t'on besoin de la domainTransaction ici?
```suggestion
  completeByAssessmentId({knexTransaction}, assessmentId) {
    return this.updateStateById(knexTransaction, { id: assessmentId, state: Assessment.states.COMPLETED });
```

Reply:
Techniquement oui, c'est le même niveau de couplage mais dans la signature j'aime bien nommer l'abstraction (`domainTransaction`) plutôt que ce qu'elle est censé contenir. J'aimerai encore mieux, qu'on aie une vérification de type mais ça on n'a pas.